src/api.rs

```diff
@@ -33,16 +33,110 @@ pub struct EncoderConfig {
   pub key_frame_interval: u64,
   pub low_latency: bool,
   pub quantizer: usize,
-  pub speed: usize,
-  pub tune: Tune
+  pub tune: Tune,
+  pub min_block_size: BlockSize,
+  pub multiref: bool,
+  pub fast_deblock: bool,
+  pub reduced_tx_set: bool,
+  pub tx_domain_distortion: bool,
+  pub encode_bottomup: bool,
+  pub rdo_tx_decision: bool,
+  pub prediction_modes: PredictionModesSetting,
+  pub include_near_mvs: bool,
 }
 
 impl Default for EncoderConfig {
   fn default() -> Self {
-    EncoderConfig { key_frame_interval: 30, low_latency: true, quantizer: 100, speed: 0, tune: Tune::Psnr  }
+    const DEFAULT_SPEED: usize = 3;
+    Self::with_speed_preset(DEFAULT_SPEED)
   }
 }
 
+impl EncoderConfig {
+  pub fn with_speed_preset(speed: usize) -> Self {
+    let mut cfg = EncoderConfig {
+      key_frame_interval: 30,
+      low_latency: true,
+      quantizer: 100,
+      tune: Tune::Psnr,
+      ..Default::default()
+    };
+    cfg.apply_speed_preset(speed);
+    cfg
+  }
+
+  pub fn apply_speed_preset(&mut self, speed: usize) {
+    self.min_block_size = Self::min_block_size_preset(speed);
+    self.multiref = Self::multiref_preset(speed);
+    self.fast_deblock = Self::fast_deblock_preset(speed);
+    self.reduced_tx_set = Self::reduced_tx_set_preset(speed);
+    self.tx_domain_distortion = Self::tx_domain_distortion_preset(speed);
+    self.encode_bottomup = Self::encode_bottomup_preset(speed);
+    self.rdo_tx_decision = Self::rdo_tx_decision_preset(speed);
+    self.prediction_modes = Self::prediction_modes_preset(speed);
+    self.include_near_mvs = Self::include_near_mvs_preset(speed);
+  }
+
+  fn min_block_size_preset(speed: usize) -> BlockSize {
+    if speed <= 1 {
+      BlockSize::BLOCK_4X4
+    } else if speed <= 2 {
+      BlockSize::BLOCK_8X8
+    } else if speed <= 3 {
+      BlockSize::BLOCK_16X16
+    } else if speed <= 4 {
+      BlockSize::BLOCK_32X32
+    } else {
+      BlockSize::BLOCK_64X64
+    }
+  }
+
+  fn multiref_preset(speed: usize) -> bool {
+    speed <= 2
+  }
+
+  fn fast_deblock_preset(speed: usize) -> bool {
+    speed > 3
```

Comment:
Just, as preference, can you keep using <= and >=, so that inclusive? I.e. speed >= 4
(Sorry- I could not be picky here)